doc/TAM/SAI-Proposal-TAM-stream-telemetry.md

```diff
@@ -0,0 +1,967 @@
+        SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_STATS_MODE SAI_STATS_MODE_READ "Set read counter mode"
+    }
+    TAM[TAM] {
+        SAI_ID SAI_VALUE "Comments"
+        SAI_TAM_ATTR_TELEMETRY_OBJECTS_LIST sai_tam_telemetry_obj
+        SAI_TAM_ATTR_TAM_BIND_POINT_TYPE_LIST SAI_TAM_BIND_POINT_TYPE_PORT
+    }
+    switch[Switch] {
+        SAI_ID SAI_VALUE "Comments"
+        SAI_SWITCH_ATTR_TAM_OBJECT_ID sai_tam_obj
+        SAI_SWITCH_ATTR_TAM_BUFFER_SIZE buffer_size
+        SAI_SWITCH_ATTR_TAM_CHUNK_SIZE chunk_size
+        SAI_SWITCH_ATTR_TAM_TELEMETRY_STATE_CHANGE_NOTIFY on_tam_telemetry_state_change
+    }
+
+    host_table_entry |o--|| hostif: binds
+    host_table_entry |o--|| hostif_trap: binds
+    hostif_trap |o--|| hostif_trap_group: binds
+    collector |o--|| hostif_trap: binds
+    collector |o--|| transport: binds
+    telemetry_type |o--|| report: binds
+    telemetry |o--o| telemetry_type: binds
+    telemetry }o--o{ collector: binds
+    counter_subscription }o--|| telemetry_type: binds
+    TAM |o--o{ telemetry: binds
+    switch |o..o| TAM: binds
+```
+
+| Object Type              | Scope                                 |
+| ------------------------ | ------------------------------------- |
+| HOSTIF                   | Global                                |
+| HOSTIF_trap_group        | Global                                |
+| HostIF_user_defined_trap | Global                                |
+| HOSTIF_table_entry       | Global                                |
+| TAM_transport            | Global                                |
+| TAM_collector            | Global                                |
+| TAM                      | Global                                |
+| TAM_telemetry            | per profile                           |
+| TAM_telemetry_type       | per profile / per type of one profile |
+| TAM_report               | per profile / per type of one profile |
+| TAM_counter_subscription | per stats of object                   |
+
+#### TAM Telemetry state machine
+
+``` mermaid
+
+---
+title: TAM Telemetry state machine
+---
+stateDiagram-v2
+    direction LR
+
+    state "Stop TAM Stream" as stop
+    state "Start TAM Stream" as start
+    state "Commit configure" as commit
+    state "Config ready" as ready
+    state join_commit <<join>>
+
+    [*] --> stop: Object create
+    stop --> start: NOS set
+    stop --> join_commit: NOS set
+
+    start --> join_commit: NOS set (The record stream will not stop and original templates will still be used)
+    join_commit --> commit: Vendor start preparing templates based on bounded telemetry types
+    start --> stop: NOS set
+
+    commit --> ready: Vendor set (callback via notification)
+
+    ready --> start: NOS set (New templates will be applied to generate the new record streaming)
+    ready --> stop: NOS set
+
+    stop --> [*]: Object remove
+
+```
+
+#### Initialize TAM cache count for Switch

```

Comment:
looks like we are probably missing the mix-mode capability query here. is it because these 2 modes are being merged together?

Reply:
https://github.com/opencomputeproject/SAI/pull/2089/files#diff-0e0ad14d84f3332634521f6b2d53007b4f7ccec6c53e003139ef77da1a4d9d50R680

This one?

Reply:
Done